profiling/proto/service/service.proto

```diff
@@ -0,0 +1,87 @@
+// Copyright 2019 gRPC authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+option go_package = "google.golang.org/grpc/profiling/proto/service";
+
+// EnableRequest defines the fields in a /Profiling/Enable method request to
+// toggle profiling on and off within a gRPC program.
+message EnableRequest {
+	// Setting this to true will enable profiling. Setting this to false will
+	// disable profiling.
+	bool enabled = 1;
+}
+
+// EnableResponse defines the fields in a /Profiling/Enable method response.
+message EnableResponse {
+}
+
+// GetStreamStats defines the fields in a /Profiling/GetStreamStats method
+// request to retrieve stream-level stats in a gRPC client/server.
+message GetStreamStatsRequest {
+}
+
+// Equivalent to google.golang.org/grpc/internal/profiling.Timer, but defined
+// for proto. Each TimerProto measures the start and end of execution of a
+// component within gRPC that's being profiled.
+message TimerProto {
+	// timer_tag is an arbitrary string that is used to tag a timer for
+	// identification purposes later.
+	string timer_tag = 1;
+
+	// begin_sec and begin_nsec are the start epoch second and nanosecond,
+	// respectively, of the component profiled by this timer in UTC. begin_nsec
+	// must be a non-negative integer.
+	int64 begin_sec = 2;
+	int32 begin_nsec = 3;
+
+	// end_sec and end_nsec are the end epoch second and nanosecond,
+	// respectively, of the component profiled by this timer in UTC. end_nsec
+	// must be a non-negative integer.
+	int64 end_sec = 4;
+	int32 end_nsec = 5;
+
+	// go_id is the goroutine ID of the component being profiled.
+	int64 go_id = 6;
+}
+
+// Equivalent to google.golang.org/grpc/internal/profiling.Stat, but defined
```

Comment:
Not exactly related to this PR. But should the internals of profiling live in the top-level `internal` directory as you have mentioned in this comment, or should they moved to an `internal` directory within `profiling`?

Reply:
Agreed. The second one looks nicer.